### PR TITLE
Remove redundant arguments in project

### PR DESCRIPTION
**Pull request by PercyDan54**
Haven't looked into it in deep, but seems ok

### CODE REVIEW

File: osu.Game.Tests/NonVisual/TaskChainTest.cs [hunk 1]
```diff
@@ -32,25 +32,25 @@ public void TearDown()
         [Test]
         public async Task TestChainedTasksRunSequentially()
         {
-            var task1 = addTask();
+            var (task1, mutex, _) = addTask();
```
Comment (peppy) [inline osu.Game.Tests/NonVisual/TaskChainTest.cs:35]: I think this reads worse than what we had.

File: osu.Game/Beatmaps/Beatmap.cs [hunk 2]
```diff
@@ -55,7 +55,7 @@ public double GetMostCommonBeatLength()
             // Note: This is more accurate and may present different results because osu-stable didn't have the ability to calculate slider durations in this context.
             double lastTime = HitObjects.LastOrDefault()?.GetEndTime() ?? ControlPointInfo.TimingPoints.LastOrDefault()?.Time ?? 0;
 
-            var mostCommon =
+            var (beatLength, duration) =
```
Comment (peppy) [inline osu.Game/Beatmaps/Beatmap.cs:58]: lost context

File: osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs [hunk 3]
```diff
@@ -87,7 +87,7 @@ private void handleEvents(string line)
                     break;
             }
 
-            line = line.Substring(depth);
+            line = line[depth..];
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs:90]: not sure we want this

File: osu.Game/Database/ArchiveModelManager.cs [hunk 4]
```diff
@@ -722,7 +722,7 @@ private IEnumerable<(string original, string shortened)> getShortenedFilenames(A
 
             // import files to manager
             foreach (string file in reader.Filenames)
-                yield return (file, file.Substring(prefix.Length).ToStandardisedPath());
+                yield return (file, file[prefix.Length..].ToStandardisedPath());
```
Comment (peppy) [inline osu.Game/Database/ArchiveModelManager.cs:725]: again, not sure.

File: osu.Game/Database/UserLookupCache.cs [hunk 5]
```diff
@@ -61,17 +61,17 @@ private void performLookup()
             {
                 while (pendingUserTasks.Count > 0 && userTasks.Count < 50)
                 {
-                    (int id, TaskCompletionSource<User> task) next = pendingUserTasks.Dequeue();
+                    (int id, TaskCompletionSource<User> task) = pendingUserTasks.Dequeue();
```
Comment (peppy) [inline osu.Game/Database/UserLookupCache.cs:64]: lost context

File: osu.Game/IO/Legacy/SerializationReader.cs [hunk 6]
```diff
@@ -241,7 +241,7 @@ public override Type BindToType(string assemblyName, string typeName)
                         {
                             if (typ.Contains("Version"))
                             {
-                                string asmTmp = typ.Substring(typ.IndexOf(',') + 1);
+                                string asmTmp = typ[(typ.IndexOf(',') + 1)..];
```
Comment (peppy) [inline osu.Game/IO/Legacy/SerializationReader.cs:244]: jesus

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/IO/Legacy/SerializationReader.cs (hunk 6 above)
Comment (PercyDan54) [reply osu.Game/IO/Legacy/SerializationReader.cs:244]: So shall we disable this inspection? Because now it's an error level thing.

Comment (peppy) [reply osu.Game/IO/Legacy/SerializationReader.cs:244]: it was never enabled. your IDE is broken; fix your local settings.

Comment (PercyDan54) [reply osu.Game/IO/Legacy/SerializationReader.cs:244]: https://github.com/ppy/osu/blob/01961fe4f9739d161f35b8b7c69e2de631f96c3e/.editorconfig#L171
It was

Comment (peppy) [reply osu.Game/IO/Legacy/SerializationReader.cs:244]: What is throwing an error from that?

Comment (PercyDan54) [reply osu.Game/IO/Legacy/SerializationReader.cs:244]: Visual Studio
image

Comment (peppy) [reply osu.Game/IO/Legacy/SerializationReader.cs]: Please remove it then, yes.

File: osu.Game.Rulesets.Mania/Edit/ManiaBeatSnapGrid.cs [hunk 7]
```diff
@@ -107,21 +107,21 @@ private void createLines()
             if (selectionTimeRange == null)
                 return;
 
-            var range = selectionTimeRange.Value;
+            var (start, end) = selectionTimeRange.Value;
```
Comment (bdach) [inline osu.Game.Rulesets.Mania/Edit/ManiaBeatSnapGrid.cs:110]: This is also arguably worse what with the `start - time` expressions below. Should be `(rangeStart, rangeEnd)` if we *really* want the deconstruction imo. Which I'm not super sold on in the first place.

File: osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapEncoderTest.cs [hunk 8]
```diff
@@ -40,13 +40,13 @@ public class LegacyBeatmapEncoderTest
         public void TestEncodeDecodeStability(string name)
         {
             var decoded = decodeFromLegacy(beatmaps_resource_store.GetStream(name), name);
-            var decodedAfterEncode = decodeFromLegacy(encodeToLegacy(decoded), name);
+            var (beatmap, beatmapSkin) = decodeFromLegacy(encodeToLegacy(decoded), name);
```
Comment (bdach) [inline osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapEncoderTest.cs:43]: arguably worse than before

Comment (PercyDan54) [reply osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapEncoderTest.cs:43]: Yeah acutually after I read the diff... I probably prefer to revert that and again disable the inspection

**Comment by PercyDan54** [comment]
Whoops the commit name was wrong... It disables IDE0042

**Comment by peppy** [comment]
Please reopen a change which fixes your IDE to no longer complain about any of the warnings.

If you want to apply code quality changes, please open one PR per (batch of) changes.

**Pull request closed without merging**